doc/src/compute_pair_local.txt

```diff
@@ -62,7 +62,10 @@ pair styles do not define any additional quantities, so N = 0.  An
 example of ones that do are the "granular pair styles"_pair_gran.html
 which calculate the tangential force between two particles and return
 its components and magnitude acting on atom I for N = 1,2,3,4.  See
-individual pair styles for details.
+individual pair styles for details. When using hybrid pair styles,
```

Comment:
I think this sentence is confusing.  How about something like this:

If one of the "pair_style hybrid"_pair_hybrid.html commands is being used,
then {pN} will be the sum of contributions from any pair sub-styles which define it
and which are active for the particular pairwise interaction (based on the atom
types of the 2 atoms).

Reply:
this is not correct. it never was the sum (things were just copied over) and it cannot be the sum, since the different pair styles may store very different properties in that vector and particularly those may be properties that may not make sense if summed. the modified code will contain the "combined" list of properties, i.e. those of the first pair style, then followed by the second (active) pair style and so on and then padded with zeros in case not all pair styles are active for a given pair of atom types.

Reply:
Then I'm still confused what you mean by combined and padding.  Say the user specifies p2, and
there are 4 sub-styles in the PH comand, only 3 of which apply to types 1,2, and only 2 of the 3 compute a p2.  Then what is the output?  4 values, 3 values, some of them zero, which ones? I agree it's complicated.  I just don't think any of these doc sentences are clear to the user.  Maybe an example or 2 is needed, so it's not just words.

Reply:
There are two problems: a) pair style hybrid and hybrid/overlay use the same function and b) you cannot just sum the output, since those could be very different properties. the unchanged code just copies properties over, but only up to the minimum number of extra properties. The more i think about it, the more it seems better to change a) and have a different function for hybrid and hybrid/overlay. For style hybrid the data from whatever the active pair style is would be copied over and then padded with zeros so that the number of total entries is the maximum of whatever the different pair styles support. if a pair style has no added data those would be all zeros.  for style hybrid/overlay, the output would be the concatenated list of all vectors from all pair styles (i.e. its length would be the sum of the lengths of the individual pair styles) and the data would be replaced by zeros, if a pair style is not active for a given pair of atom types. Right now, it outputs only the active pair styles, so for hybrid/overlay outputting p2 would be p2 of the first active pair style, thus it could be a different property depending on which pair styles are active. for hybrid, the output would be the same, i.e. p2 would be p2 for whatever pair style is active, since there can only be one active pair style.

Reply:
Hmmm - this is getting more complicated.  What if we augment the syntax of the requested field, something like p2/n, where n = the index (1 to N) of the pair hybrid sub-style.  I think we've done something similar for other commands that work with pair hybrid sub-styles?

In this case it would mean the p2 value of the Nth sub-style.  Make it an error if that sub-style does not provide a p2 (can we detect that?)  Use a value of 0.0 if the sub-style is not defined for the atom types of I,J (so that the user would just see non-zero and zero values in a big list of compute pair/local output).  If the user wants to get multiple values from a sub-style, they can just list multiple fields in the pair/local command, e.g. p2/1, p2/2, p2/3.  This way a single field always means a single value.  No padding, no difference between hybrid and hybrid/overlay.

Reply:
i don't see how this can be done without an ugly hack in compute pair/local, i.e. that compute pair/local has to detect, whether it is accessing a hybrid or hybrid/local pair style and then use the more complex syntax. the implementation in pair style hybrid or hybrid/overlay could then be set up uniformly as i had outlined it for pair style hybrid/overlay and compute pair local would need to query the individual substyles of pair style hybrid, how many extra data elements the single function would provide.
considering, that this is quite a corner case as it is, and that there are very few pair styles that support this feature, i would leave compute pair/local alone and have people count out how many entries there are. the one person on lammps-users, that what reporting the bug would be ok with having the various lists of properties appended to each other